snuba/snuba_migrations/querylog/0004_add_bytes_scanned.py

```diff
@@ -0,0 +1,55 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Array, Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Adds fields for query profile.
+    """
+
+    blocking = True
+
+    def __forward_migrations(
+        self, table_name: str
+    ) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.QUERYLOG,
+                table_name=table_name,
+                column=Column(
+                    "clickhouse_queries.bytes_scanned",
+                    Array(
+                        UInt(64),
+                        Modifiers(
+                            default="arrayResize([0], length(clickhouse_queries.sql))"
+                        ),
+                    ),
```

Comment:
This will not be this simple.
If I applied this migration the way it is, the consumer would break.
The consumer would try write rows in the `clickhouse_queries` nested column that would not be of the same size as the `bytes_scanned` array would not be passed.
As such Clickhouse will consider this array as passed empty.

The migration has to happen in the consumer first passing the column first. You will have to set the `input_format_skip_unknown_fields` Clickhouse settings in your write so Clickhouse will ignore the unknown column. Then we can apply the migration.

This was the example https://github.com/getsentry/snuba/pull/2232/files